store/etcd/rulesets_internal_test.go

```diff
@@ -189,3 +192,64 @@ func TestCompareSignature(t *testing.T) {
 		require.Equal(t, fmt.Sprintf("invalid %s with value '%s': %s", exp.Field, exp.Value, exp.Reason), exp.Error())
 	})
 }
+
+func BenchmarkProtoMarshalling(b *testing.B) {
+	rs, err := regula.NewBoolRuleset(
+		rule.New(rule.And(rule.Not(rule.BoolValue(false)), rule.BoolParam("param")), rule.BoolValue(true)),
+		rule.New(rule.And(rule.BoolParam("1st-param"), rule.BoolParam("2nd-param")), rule.BoolValue(false)),
+	)
+	require.NoError(b, err)
+
+	pbrs := toProtobufRuleset(rs)
+	for n := 0; n < b.N; n++ {
+		_, err := proto.Marshal(pbrs)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkJSONMarshalling(b *testing.B) {
+	rs, err := regula.NewBoolRuleset(
+		rule.New(rule.And(rule.Not(rule.BoolValue(false)), rule.BoolParam("param")), rule.BoolValue(true)),
+		rule.New(rule.And(rule.BoolParam("1st-param"), rule.BoolParam("2nd-param")), rule.BoolValue(false)),
+	)
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		_, err := json.Marshal(rs)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkProtoUnmarshalling(b *testing.B) {
+	rs, err := regula.NewBoolRuleset(
+		rule.New(rule.And(rule.Not(rule.BoolValue(false)), rule.BoolParam("param")), rule.BoolValue(true)),
+		rule.New(rule.And(rule.BoolParam("1st-param"), rule.BoolParam("2nd-param")), rule.BoolValue(false)),
+	)
+	require.NoError(b, err)
+
+	bb, err := proto.Marshal(toProtobufRuleset(rs))
+	require.NoError(b, err)
+
+	var pbrs pb.Ruleset
```

Comment:
You need to move this variable inside the loop, always make sure to create a new variable when benchmarking encoding and decoding

Reply:
Done in https://github.com/heetch/regula/pull/86/commits/bd1632056f5ee3d36049cc3b9fea6fd7ccf02681.